orchard-provider-aws/src/main/scala/com/salesforce/mce/orchard/io/aws/resource/Ec2Resource.scala

```diff
@@ -104,6 +104,49 @@ case class Ec2Resource(name: String, spec: Ec2Resource.Spec) extends ResourceIO
     resultSet.nonEmpty
   }
 
+  final val GoodSummaryStatuses = Set(SummaryStatus.INITIALIZING, SummaryStatus.OK)
+
+  /**
+   * when ssm does not yet have visibility to ec2, it could be initializing phase of the ec2 instance,
+   * or it could the ec2 has instanceStatus impaired.
+   * getStatus() performs this check to fail early in case the initialized ec2 instance status has issues,
+   * and the ResourceInstance logic can move on to the next attempt if available
+   * @param ec2InstanceId  ec2 instance id
+   * @return
+   */
+  def describeEc2Status(ec2InstanceId: String): Either[Throwable, Status.Value] = {
+    val client = Client.ec2()
+    val request = DescribeInstanceStatusRequest.builder().instanceIds(ec2InstanceId).build()
+    val response = client.describeInstanceStatus(request)
+    client.close()
+    val status = response.instanceStatuses().asScala
+    status.isEmpty match {
```

Comment:
looks like you could do something like this:

```
status.toList() match {
  case sts :: _ => // your case _, but no longer need to call `status.head` anymore
  case _ => // for your case true
}
```

Reply:
Got it.  new commit pushed.